addons/crm/models/crm_lead.py

```diff
@@ -1860,7 +1860,12 @@ def _message_partner_info_from_emails(self, emails, link_mail=False):
         "Raoul" <raoul@raoul.fr> as recipient. """
         result = super(Lead, self)._message_partner_info_from_emails(emails, link_mail=link_mail)
         for email, partner_info in zip(emails, result):
-            if partner_info.get('partner_id') or not email or not (self.partner_name or self.contact_name):
+            if (
+                partner_info.get('partner_id')
+                or not email
+                or not (self.partner_name or self.contact_name)
+                or not self.email_from
```

Comment:
Technically we could check this once and return early. It should be the same for (partner_name or contact_name)

Since the diff is comparable either way I'd do that.